### ISSUE TITLE
Add Extension System

### ISSUE THREAD

**Issue by RiedleroD** [asuender/cliex#4]

**Comment by mfederczuk** [comment]
We could use [Lua](https://www.lua.org) as a scripting language for the plugins

**Comment by RiedleroD** [comment]
No. Plugins are meant to be outside classes with standard functions like fileopen() and init() or something. I'm going to make it in a seperate branch and you can tell me what you think of it.
Also I don't like lua. Had way too many problems with bad lua support in my life so I advise against using lua in general.

**Comment by mfederczuk** [comment]
So plugins are supposed to be C/C++ code, added in at compile time?

**Comment by RiedleroD** [comment]
Well I planned to somehow import the plugins at runtime but I searched and found nothing so compile time it is, yes.
Maybe we can work in lua AND compiletime plugins. Because I hate the idea of a lua-based music player but also hate the idea of not having a plugin system that loads plugins at runtime.

**Comment by mfederczuk** [comment]
Maybe rename the compile time plugins to "extensions". Makes it less ambiguous

**Comment by RiedleroD** [comment]
That's probably needed, yes.
On another note, you can code in Lua, right? I'd make another issue for the Lua plugin system and assign you to it.

**Comment by mfederczuk** [comment]
I read through the Lua page in [learnxinyminutes.com](https://learnxinyminutes.com/docs/lua), and tested out the Lua C API, so yeah, I could write the Lua plugin system

**Comment by RiedleroD** [comment]
#19 it is
